### DISCUSSION TITLE
Add support for image transformations in bulk storage URL signing (getSignedUrls)

### DISCUSSION THREAD

**Discussion by vtsatskin** [supabase/supabase#28893] [Feature Requests]
At the time of writing, Supabase storage supports creating temporary signed URLs with image  transformations applied for **single** image objects. This is documented officially in the following places:

1. https://supabase.com/docs/reference/javascript/storage-from-createsignedurl
1. https://supabase.com/docs/guides/storage/serving/image-transformations?queryGroups=language&language=js#signing-urls-with-transformation-options

However, there is currently no way to sign a list of URLs with image transformation options. So:
* `createSignedUrl` supports image transformations
* `createSignedUrls` does NOT support image transformations

Based on community feedback, I'm not the only one who has run into this use case not being supported. Please see https://github.com/supabase/supabase/issues/15343 dating back to June 2023. 

It would be great to see this implemented. Based on my research, it should be a trivial feature request since the functionality already exists for `createSignedUrl`. From my understanding, an update to the supabase/storage-js client library would need to be made and supabase/storage server would need to be made.

For someone implementing, please take a look at my comment for a partial investigation into the scope of the changes needed: https://github.com/supabase/supabase/issues/15343#issuecomment-2310691270

**Comment by mikkelam** [comment]
Also quite interested in this
